src/Integration.UnitTests/Binding/ProjectBinderFactoryTests.cs

```diff
@@ -102,10 +131,12 @@ public void Get_CppProject_CFamilyProjectBinderReturned()
         }
 
         [TestMethod]
-        public void Get_NonRoslynProject_CFamilyProjectBinderReturned()
+        public void Get_UnknownProject_CFamilyProjectBinderReturned()
         {
-            var project = new ProjectMock("c:\\foo.proj");
-            project.ProjectKind = "{" + Guid.NewGuid() + "}";
+            // todo: this is wrong and should be null
```

Comment:
This sounds like a bug. Is there a ticket to track it?

Reply:
I think somewhere along the path it works out, because we filter supported projects before calling this code (based on language). And since we support only roslyn and cpp ATM, if we got to this point and it's not roslyn, then it "must be" cpp. So this fallback was technically working.
That said, it's obviously wrong and if we support JS it would be broken. But at the moment "it works".

Reply:
Ok. This isn't a new test, but it's clearly the wrong behaviour. If the language is unknown / not Roslyn/CFamily the factory should return null.

Having a test that checks the behaviour is wrong isn't very useful or discoverable.
It would be better to have an [Ignored] test that checks for the desired behaviour and fails, ideally with a ticket to fix it.